improver/utilities/spatial.py

```diff
@@ -302,43 +302,49 @@ def __init__(self, regrid=False):
         self.regrid = regrid
 
     @staticmethod
-    def _update_metadata(gradient, cube):
-        """Rename cube and remove inappropriate metadata.
+    def _update_metadata(gradient, cube, diff, axis):
+        """Rename cube and assign appropriate units.
 
         Args:
             gradient (iris.cube.Cube)
+                Cube containing gradients along the x or y axis.
             cube (iris.cube.Cube)
+                Cube used to calculate differences and gradients.
+            diff (iris.cube.Cube)
+                Cube containing differences along the x or y axis.
+            axis (str)
 
         Returns:
             iris.cube.Cube:
                 Gradient cube with updated metadata.
 
         """
         gradient.rename("gradient_of_" + cube.name())
+        gradient.units = cube.units / diff.coord(axis=axis).units
         return gradient
 
-    def _gradient_from_diff(self, diff_cube, ref_cube, coord_axis):
+    def _gradient_from_diff(self, diff, cube, axis):
         """
         Calculate the gradient along the x or y axis from differences between
         adjacent grid squares.
 
         Args:
-            diff_cube (iris.cube.Cube):
+            diff (iris.cube.Cube):
                 Cube containing differences along the x or y axis
-            ref_cube (iris.cube.Cube):
+            cube (iris.cube.Cube):
                 Cube with correct output dimensions
-            coord_axis (str):
+            axis (str):
```

Comment:
Why define axis in one private method but not the other? Add to the other perhaps for consistency?

Reply:
I've defined it in both private methods for consistency.